docs: update custom domain docs (#19865)
supabase/supabase: apps/docs/pages/guides/platform/custom-domains.mdx

  ></iframe>
</div>

There are two types of customizable domains supported by Supabase:

1. Custom domains, where you use a domain such as `api.example.com` instead of the project's default domain.

> 2. Vanity subdomains (experimental), where you can setup a different Subdomain on `supabase.co` for your project.

Reviewer: ```suggestion
2. Vanity subdomains (experimental), where you can set up a different subdomain on `supabase.co` for your project.
```

can you give an example for this one?

> You can only configure one or the other at this time.

Reviewer: ```suggestion
You can choose either a custom domain or vanity subdomain for each project.
```

I'm guessing this is scoped by project? (not sure, feel free to fix)

## Custom domains

Custom domains allow you to change the way your project's URLs appear to your users. This is useful when:

- You are using [OAuth (Social Login)](/docs/guides/auth/social-login) with Supabase Auth and the project's URL is shown on the OAuth consent screen.

> - You are creating APIs for third-party systems, like when implementing webhooks or external API calls to your project via [Edge Functions](/docs/guides/functions).

Reviewer: ```suggestion
- You are creating APIs for third-party systems, for example, implementing webhooks or external API calls to your project via [Edge Functions](/docs/guides/functions).
```

- You are storing URLs in a database or encoding them in QR codes.

> Configuring a custom domain also helps you keep your APIs portable for the long-term. By using a custom domain you can easily migrate from one Supabase project to another, or make it easier to version APIs in the future.

Reviewer: `long term` without the hyphen

### Configure a custom domain using the Supabase dashboard

Follow the **Custom Domains** steps in the [General Settings](https://supabase.com/dashboard/project/_/settings/general) page in the Dashboard to set up a custom domain for your project.

### Configure a custom domain using the Supabase CLI

This example assumes your Supabase project is `abcdefghijklmnopqrst` with a corresponding API URL `abcdefghijklmnopqrst.supabase.co` and configures a custom domain at `api.example.com`.

To get started:

1. [Install](/docs/guides/resources/supabase-cli) the latest version of the Supabase CLI.
2. [Log in](/docs/guides/resources/supabase-getting-started/local-development#log-in-to-the-supabase-cli) to your Supabase account using the CLI.
3. Setting up a custom domain requires [Owner or Admin permissions](/docs/guides/platform/access-control#manage-team-members) for the project.

> 4. Get a custom domain. Currently, only subdomains are supported.

Reviewer: ```suggestion
4. Get a custom domain from a DNS provider. Currently, only subdomains are supported.
```

flexible on the text, just want to make it clear this part has to be done with someone else and isn't something you configure on our dashboard

   - Use `api.example.com` instead of `example.com`.

### Add a CNAME record

> To make sure your custom domain points to the Supabase project you need to add a CNAME record to your domain's DNS settings.

Reviewer: ```suggestion
Add a CNAME record to your domain's DNS settings that points your custom domain to your Supabase project URL.
```

If your project's default domain is `abcdefghijklmnopqrst.supabase.co` you should:

- Create a CNAME record for `api.example.com` that resolves to `abcdefghijklmnopqrst.supabase.co.`.
- Use a low TTL value so that you can see changes propagating quickly if you make a mistake.

### Verify ownership of the domain

> In order for Supabase to know that you're the rightful owner of the domain, you need to add special TXT records that prove your ownership of it.

Reviewer: ```suggestion
Register your domain with Supabase to prove that you own it. You need to download two TXT records and add them to your DNS settings.
```

> Use the [`domains create`](/docs/reference/cli/supabase-domains-create) command of the CLI to register the domain with Supabase and retrieve TXT verification records:

Reviewer: ```suggestion
In the CLI, run [`domains create`](/docs/reference/cli/supabase-domains-create) to register the domain and Supabase and get your verification records:
```

```bash
supabase domains create --project-ref abcdefghijklmnopqrst --custom-hostname api.example.com
```

> Two TXT records are returned and you will need to add them to the domain's DNS settings. For example:

Reviewer: ```suggestion
Two TXT records are returned. For example:
```

```text
[...]
Required outstanding validation records:
        _cf-custom-hostname.api.example.com TXT -> 46BBC14D-D50A-409C-8DB5-F862CF5BA660
        api.example.com TXT -> ca3-F1HvR9i938OgVwpCFwi1jTsbhe1hvT0Ic3efPY3Q
```

> When adding the TXT records make sure you remove any extra spaces surrounding the values. Use a low TTL value so you can quickly change them if you make a mistake.

Reviewer: ```suggestion
Add the records to your domains' DNS settings. Make sure to trim surrounding whitespace. Use a low TTL value so you can quickly change the records if you make a mistake.
```

### Verify your domain

> Once you have setup the DNS records:

Reviewer: ```suggestion
Make sure you've configured all required DNS settings:
```

- CNAME for your custom domain pointing to the Supabase project domain.
- TXT record for `_cf-custom-hostname.<domain>`.
- TXT record for `<domain>`.

> Use the [`domains reverify`](/docs/reference/cli/supabase-domains-reverify) command to begin the verification process of your domain. You may need to run this command a few times as DNS records take a while to propagate.

Reviewer: ```suggestion
Use the [`domains reverify`](/docs/reference/cli/supabase-domains-reverify) command to begin the verification process of your domain. You may need to run this command a few times because DNS records take a while to propagate.
```

```bash
supabase domains reverify --project-ref abcdefghijklmnopqrst
```

> In the background, Supabase will check your DNS records and use [Let's Encrypt](https://letsencrypt.org) to issue a SSL certificate for your domain. This process can take a while, sometimes up to 30 minutes.

Reviewer 2: ```suggestion
In the background, Supabase will check your DNS records and use [Let's Encrypt](https://letsencrypt.org) to issue a SSL certificate for your domain. This process can take up to 30 minutes.
```

Reviewer: ```suggestion
In the background, Supabase checks your DNS records and uses [Let's Encrypt](https://letsencrypt.org) to issue a SSL certificate for your domain. This process can take up to 30 minutes.
```

### Prepare to activate your domain

> Before you activate your domain, you need to understand the consequences and behavior of your Supabase project.

Reviewer: ```suggestion
Before you activate your domain, prepare your applications and integrations for the domain change:
```

> - The project's Supabase domain remains active and will not go away.

Reviewer 2: ```suggestion
- The project's Supabase domain remains active
```

>   - You do not need to change the Supabase URL in your applications immediately or at once.

Reviewer 2: ```suggestion
  - You do not need to change the Supabase URL in your applications immediately

  - You can use it interchangeably with the custom domain.
- Supabase Auth will use the custom domain immediately once activated.
  - OAuth flows will advertize the custom domain as a callback URL.
  - SAML will use the custom domain instead. This means that the `EntityID` of your project has changed, and this may cause SAML with existing identity providers to stop working.

> To prevent issues for your users, make sure you have gone through these steps:

Reviewer: ```suggestion
To prevent issues for your users, follow these steps:
```

> 1. Go through all of your Supabase OAuth providers:

Reviewer: ```suggestion
1. For each of your Supabase OAuth providers:
```

>    - In the provider's developer console (not in the Supabase dashboard!), find the OAuth application and add the custom domain Supabase Auth callback URL **in addition to the Supabase project URL.** Example:

Reviewer: ```suggestion
   - In the provider's developer console (not in the Supabase dashboard), find the OAuth application and add the custom domain Supabase Auth callback URL **in addition to the Supabase project URL.** Example:
```

     - `https://abcdefghijklmnopqrst.supabase.co/auth/v1/callback` **and**
     - `https://api.example.com/auth/v1/callback`

>    - [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) uses cookies bound to the project's domain. In this case make sure your frontend code uses the custom domain instead of the default project's domain.

Reviewer: ```suggestion
   - [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) uses cookies bound to the project's domain. Make sure your frontend code uses the custom domain instead of the default project's domain.
```

> 2. Go through all of your SAML identity providers:

Reviewer: ```suggestion
2. For each of your SAML identity providers:
```

>    - You will need to reach out via email to all of your existing identity providers and ask them to update the metadata for the SAML application (your project). Use `https://api.example.com/auth/v1/...` instead of `https://abcdefghijklmnopqrst.supabase.co/auth/v1/callback`.

Reviewer: ```suggestion
   - Contact your provider and ask them to update the metadata for the SAML application. They should use `https://api.example.com/auth/v1/...` instead of `https://abcdefghijklmnopqrst.supabase.co/auth/v1/callback`.
```

   - Once these changes are made, SAML Single Sign-On will likely stop working until the domain is activated. Plan for this ahead of time.

### Activate your domain

Once you've done the necessary preparations to activate the new domain for your project, you can activate it using the [`domains activate`](/docs/reference/cli/supabase-domains-activate) CLI command.

```bash
supabase domains activate --project-ref abcdefghijklmnopqrst
```

When this step completes, Supabase will serve the requests from your new domain. The Supabase project domain **continues to work** and serve requests so you do not need to rush to change client code URLs.

> If you wish to use the new domain in client code, you can set it up like so:

Reviewer: ```suggestion
If you wish to use the new domain in client code, change the URL used in your Supabase client libraries:
```

```js
import { createClient } from '@supabase/supabase-js'

// Use a custom domain as the supabase URL
const supabase = createClient('https://api.example.com', 'public-anon-key')
```

> When using [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) make sure your frontend code is using the custom domain only!

Reviewer: ```suggestion
```

it's mentioned above so I would lean towards removing this, or perhaps throwing it into an admonition, it seems a bit out of place here

### Remove a custom domain

Removing a custom domain may cause some issues when using Supabase Auth with OAuth or SAML. You may have to reverse the changes made in the _[Prepare to activate your domain](#prepare-to-activate-your-domain)_ step above.

To remove an activated custom domain you can use the [`domains delete`](/docs/reference/cli/supabase-domains-delete) CLI command.

```bash
supabase domains delete --project-ref abcdefghijklmnopqrst
```

## Vanity subdomains

> Vanity subdomains allow you to present a basic branded experience, compared to custom domains. They allow you to host your services at a custom subdomain on Supabase (e.g., `my-example-brand.supabase.co`) instead of the default, randomly-assigned `abcdefghijklmnopqrst.supabase.co`.

Reviewer: ```suggestion
Vanity subdomains allow you to present a basic branded experience, compared to custom domains. They allow you to host your services at a custom subdomain on Supabase (e.g., `my-example-brand.supabase.co`) instead of the default, randomly assigned `abcdefghijklmnopqrst.supabase.co`.
```

1. [Log in](/docs/guides/resources/supabase-getting-started/local-development#log-in-to-the-supabase-cli) to your Supabase account using the CLI.
1. Ensure that you have [Owner or Admin permissions](/docs/guides/platform/access-control#manage-team-members) for the project you'd like to set up a vanity subdomain for.
1. Ensure that your organization is on a paid plan (Pro/Team/Enterprise plan) in the [Billing page of the Dashboard](https://supabase.com/dashboard/org/_/billing).

You can configure vanity subdomains via the CLI only.

Let's assume your Supabase project's domain is `abcdefghijklmnopqrst.supabase.co` and you wish to configure a vanity subdomain at `my-example-brand.supabase.co`.

### Check subdomain availability

Use the [`vanity-subdomains check-availability`](/docs/reference/cli/supabase-vanity-subdomains-check-availability) command of the CLI to check if your desired subdomain is available for use:

```bash
supabase vanity-subdomains --project-ref abcdefghijklmnopqrst check-availability --desired-subdomain my-example-brand --experimental
```

### Prepare to activate the subdomain

> Before you activate the subdomain, you need to understand the consequences and behavior of your Supabase project.

Reviewer: ```suggestion
Before you activate your vanity subdomain, prepare your applications and integrations for the subdomain change:
```
